components/load_from_pdf/src/main.py

```diff
@@ -73,47 +70,53 @@ def _get_meta_df() -> pd.DataFrame:
 
         return dask_df
 
-    def load(self) -> dd.DataFrame:
-        if self.protocol == "file":
-            logger.info("Found PDF files local file system")
-
-            if self.fs.exists(self.pdf_path):
-                if self.fs.isdir(self.pdf_path):
-                    pdf_dir = self.pdf_path
-                else:
-                    pdf_dir = os.path.dirname(self.pdf_path)
-
-                loader = PyPDFDirectoryLoader(pdf_dir)
-                documents = loader.load()
+    def load_pdf_from_fs(self, file_path: str):
+        with self.fs.open(file_path, "rb") as pdf_file:
+            pdf_bytes = pdf_file.read()
 
-            else:
-                msg = "PDF path does not exist"
-                raise ValueError(msg)
-
-        else:
-            logger.info("Found PDF files on remote file system")
+        documents = fitz.open("pdf", pdf_bytes)
+        # get all text
+        text = "".join([document.get_text() for document in documents])
+        documents.close()
 
-            files = self.fs.ls(self.pdf_path)
+        return text
 
-            with tempfile.TemporaryDirectory() as temp_dir:
-                for file_path in tqdm(files):
-                    if file_path.endswith(".pdf"):
-                        file_name = os.path.basename(file_path)
-                        temp_file_path = os.path.join(temp_dir, file_name)
-                        self.fs.get(file_path, temp_file_path)
+    def process_pdf(self, row):
+        file_path = row["pdf_path"]
+        text = self.load_pdf_from_fs(file_path)
+        row["file_name"] = file_path.split("/")[-1]  # Extracting filename
+        row["text"] = text
+        return row
 
-                loader = PyPDFDirectoryLoader(temp_dir)
-                documents = loader.lazy_load()
-
-        doc_dict = defaultdict(list)
-        for doc_counter, document in enumerate(documents):
-            doc_dict["file_name"].append(os.path.basename(document.metadata["source"]))
-            doc_dict["text"].append(document.page_content)
-
-            if doc_counter == self.n_rows_to_load:
-                break
+    def load(self) -> dd.DataFrame:
+        try:
+            file_paths = self.fs.ls(self.pdf_path)
+        except NotADirectoryError:
+            file_paths = [self.pdf_path]
+
+        file_paths = [
+            file_path for file_path in file_paths if file_path.endswith(".pdf")
+        ]
+
+        if self.n_rows_to_load is not None:
+            file_paths = file_paths[: self.n_rows_to_load]
+
+        dask_df = dd.from_pandas(
+            pd.DataFrame({"pdf_path": file_paths}),
+            npartitions=os.cpu_count(),
```

Comment:
This could probably be a parameter so it can be upped for larger datasets.

Reply:
Added :)